### DISCUSSION TITLE
Content Security Policy Refused to apply inline style

### DISCUSSION THREAD

**Discussion by spomar** [globaleaks/globaleaks-whistleblowing-software#3916] [Q&A] [answered]
Versione 4.13.22
In diverse pagine loggo questo blocco CSP; sto sbagliando qualcosa?
Ho rimosso qualsiasi personalizzazione css e riavviato il servizio globaleaks piu' volte.
Non loggo errori in /var/globaleaks/log/globaleaks.log

Grazie mille a chi vorra' rispondere

`scripts.min.js:8 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'sha256-pru43GdcNLwb4MwzOriCI9/9cKBzE5xeoLWHlKai1As='". Either the 'unsafe-inline' keyword, a hash ('sha256-uD+9kGdg1SXQagzGsu2+gAKYXqLRT/E07bh4OhgXN8Y='), or a nonce ('nonce-...') is required to enable inline execution. Note that hashes do not apply to event handlers, style attributes and javascript: navigations unless the 'unsafe-hashes' keyword is present.`

**Comment by evilaliv3** [comment, marked as answer]
Grazie per la domanda @spomar 

La nostra policy blocca una operazione libreria inclusa nel software ma ciò non causa alcuna problematica.

**Reply by spomar** [reply]
grazie per la risposta. nel mio specifico caso, lato javascript non funziona il menu' a tendina di ricerca utenti nella popup dedicata al trasferimento della segnalazione tra riceventi
Non parte la ricerca, non compaione risultati filtrati, e' come se lo script fosse inibito al causa di quell'errore di csp

**Reply by spomar** [reply]
Se puo' essere utile, aggiungo che la chiamata PUT parte clickando sul bottone CERCA, ma lato server loggo questa eccezione

`KeyError Mapping key not found.

Traceback (most recent call last):

  File "/usr/lib/python3/dist-packages/twisted/internet/defer.py", line 190, in maybeDeferred
    result = f(*args, **kwargs)

  File "/usr/lib/python3/dist-packages/globaleaks/rest/decorators.py", line 66, in wrapper
    return f(self, *args, **kwargs)

  File "/usr/lib/python3/dist-packages/globaleaks/rest/decorators.py", line 39, in wrapper
    return f(self, *args, **kwargs)

  File "/usr/lib/python3/dist-packages/globaleaks/rest/decorators.py", line 52, in wrapper
    return f(self, *args, **kwargs)

  File "/usr/lib/python3/dist-packages/globaleaks/handlers/operation.py", line 25, in put
    return func(self, request['args'], *args, **kwargs)

  File "/usr/lib/python3/dist-packages/globaleaks/handlers/recipient/rtip.py", line 646, in transfer_tip
    return transfer_tip_access(self.request.tid, self.session.user_id, self.session.cc, itip_id, req_args['receiver'])

KeyError: 'receiver'`

**Reply by spomar** [reply]
Mistero risolto, la ricerca fallisce perche' i miei riceventi hanno *tutti* lo stesso indirizzo email
Chiedo scusa alla community per il mio falso allarme :-(

**Comment by evilaliv3** [comment]
nessun problema, grazie anzi!

@spomar : potresti spiegarmi meglio come riprodurre la problematica così vedo di gestirla e risolverla?

**Reply by spomar** [reply]
Devo dire la verita', ho tirato a indovinare e ho risolto :-)
In pratica per una mia demo ho creato 2 utenti con ruolo di ricevente assegnando natualmente username diversi, ma stesso indirizzo email.
Questa mia scelta poco logica in produzione, mi dava l'impressione che non funzionasse la ricerca utenti nella parte in cui vuoi trasferire la segnalazione ad un collega ricevente.
Invece la ricerca funziona perfettamente, semplicemente il menu a tendina non mostra utenti con la tua casella di posta, nel mio caso erano tutti con la mia casella e quindi non tornava nulla :-D
